lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/handler/DrivingPermitHandler.java

```diff
@@ -217,21 +217,44 @@ public APIGatewayProxyResponseEvent handleRequest(
             // Driving Permit Completed Normally
             eventProbe.counterMetric(LAMBDA_DRIVING_PERMIT_CHECK_COMPLETED_OK);
             return ApiGatewayResponseGenerator.proxyJsonResponse(HttpStatusCode.OK, response);
-        } catch (OAuthHttpResponseExceptionWithErrorBody e) {
+        } catch (OAuthErrorResponseException e) {
             // Driving Permit Lambda Completed with an Error
-            LOGGER.error("Encountered error in DCS request : {}", e.getErrorReason());
-            eventProbe.counterMetric(LAMBDA_DRIVING_PERMIT_CHECK_COMPLETED_ERROR);
+
+            // Debug only as Oauth errors appear in the redirect url
+            // This will output the specific error message
+            // Note Unit tests expect server error (correctly)
+            // and will fail if logging is at debug level (during tests)
+            if (LOGGER.isDebugEnabled()) {
+                String customOAuth2ErrorDescription = e.getErrorReason();
+                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                        e.getStatusCode(), // Status Code determined by throw location
+                        new CommonExpressOAuthError(
+                                OAuth2Error.SERVER_ERROR, customOAuth2ErrorDescription));
+            }
+
+            // Non-debug route - standard OAuth2Error.SERVER_ERROR
             return ApiGatewayResponseGenerator.proxyJsonResponse(
-                    e.getStatusCode(), e.getErrorReason());
+                    e.getStatusCode(), // Status Code determined by throw location
+                    new CommonExpressOAuthError(OAuth2Error.SERVER_ERROR));
         } catch (Exception e) {
-            // Driving Permit Lambda Completed with an Error
-            LOGGER.error("Exception while handling lambda {}", context.getFunctionName());
-            eventProbe.counterMetric(LAMBDA_DRIVING_PERMIT_CHECK_COMPLETED_ERROR);
+            // This is where unexpected exceptions will reach (null pointers etc)
+            // Expected exceptions should be caught and thrown as
+            // OAuthErrorResponseException
+            // We should not log unknown exceptions, due to possibility of PII
+            LOGGER.error(
+                    "Unhandled Exception while handling lambda {} exception {}",
+                    context.getFunctionName(),
+                    e.getClass());
 
-            LOGGER.debug(e.getMessage(), e);
+            if (LOGGER.isDebugEnabled()) {
+                e.printStackTrace();
```

Comment:
Can we still use LOGGER.error("something", e) so that if were watching it an env it appears as expected

Reply:
For OAuthErrorResponseException, there must be no log line here, as this is thrown by the CRI its self. This happens when it has caught and logged an error or exception, then throws an OAuthErrorResponseException. If there is a log line here it would double log all errors in the CRI.

Reply:
I still think this needs to be LOGGER.error(e) as otherwise it wont log correctly outside of local